index.js

```diff
@@ -295,24 +279,17 @@ checkElevated()
             k8s.init();
         }
 
-        return apiClient.getAuthToken(config, pkg.name, socket);
-    })
-    .then(() => {
-        debug('got auth token:');
-        debug(config.auth_token);
         debug('connecting to log server');
-        return connectionManager.connectLogServer(config, pkg.name);
-    })
-    .then(sock => {
-        socket = sock;
+        return connectionManager.connectLogServer(config, pkg.name, cb);
+    }], (error, success) => {
```

Comment:
are we swallowing the errors here?

if `os-version` fails for example, is there just a default we fallback to? or does it not matter?

Reply:
Good point - normally, `os-version` shouldn't error if `agent` is on supported versions and also, currently, [in `promisified` version](https://github.com/logdna/logdna-agent/blob/master/index.js#L252-L253), it does nothing - I can just add log message saying there is such issue - also, [`connectLogServer`](https://github.com/logdna/logdna-agent/blob/v1.5.7/lib/connection-manager.js#L152) returns `null` as in `error`.

Reply:
Ya, it's cool to swallow it too, just maybe we need to make a note or understand what the implications are when an error does occur (particularly for os-version since we authored that file)

Reply:
The final `(error, success)` won't include `os-version`'s error since it's being used inside `async.waterfall`; therefore, adding `log` statement for the case of `error` in order to be able to investigate the issues much better.